src/server/rdb_save.cc

```diff
@@ -701,17 +707,20 @@ io::Bytes RdbSerializer::PrepareFlush() {
   return mem_buf_.InputBuffer();
 }
 
-error_code RdbSerializer::WriteJournalEntries(absl::Span<const journal::Entry> entries) {
+error_code RdbSerializer::WriteJournalEntries(const journal::Entry& entry) {
   io::BufSink buf_sink{&journal_mem_buf_};
   JournalWriter writer{&buf_sink};
-  for (const auto& entry : entries) {
-    writer.Write(entry);
-  }
+  writer.Write(entry);
 
   RETURN_ON_ERR(WriteOpcode(RDB_OPCODE_JOURNAL_BLOB));
-  RETURN_ON_ERR(SaveLen(entries.size()));
+  RETURN_ON_ERR(SaveLen(1));
   RETURN_ON_ERR(SaveString(io::View(journal_mem_buf_.InputBuffer())));
-
+  // SELECTDB is serialized inside journal writer. In rdb loader when parsing journal blob
+  // the journal reader parses the entry db index, and we dont use the main flow of
+  // updating the current db index. Unless we change the flow in the loader, we must
+  // update last_entry_db_index_ to invalid so that the next snapshot entry to be serialized
+  // will update the db index.
+  last_entry_db_index_ = kInvalidDbId;
```

Comment:
So the journal db numbering and rdb db numbering are fully independent, right? Then it seems like we don't have to reset last_entry_db_index_?